src/Geralt.Tests/XChaCha20Poly1305Tests.cs

```diff
@@ -183,4 +183,133 @@ public void Decrypt_InvalidKey()
         key = new byte[XChaCha20Poly1305.KeySize + 1];
         Assert.ThrowsException<ArgumentOutOfRangeException>(() => XChaCha20Poly1305.Decrypt(plaintext, Ciphertext, Nonce, key));
     }
+
+    [TestMethod]
+    public void IncrementalEncryptDecrypt_ValidInputs()

```

Comment:
I had to combine encryption and decryption in the same unit tests.  As the libsodium documentation states, the same sequence encrypted twice will produce different ciphertexts.